Source/SwiftLintFramework/Rules/AttributesRule.swift

```diff
@@ -0,0 +1,385 @@
+//
+//  AttributesRule.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 10/15/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+private enum AttributesRuleError: ErrorType {
+    case UnexpectedBlankLine
+    case MoreThanOneAttributeInSameLine
+}
+
+public struct AttributesRule: ASTRule, ConfigurationProviderRule {
+    public var configuration = AttributesConfiguration()
+
+    private static let parametersPattern = "^\\s*\\(.+\\)"
+
+    // swiftlint:disable:next force_try
+    private static let regularExpression = try! NSRegularExpression(pattern: parametersPattern,
+                                                                    options: [])
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "attributes_rule",
+        name: "Attributes",
+        description: "Attributes should be on their own lines in functions and types, " +
+                     "but on the same line as variables and imports.",
+        nonTriggeringExamples: AttributesRuleExamples.nonTriggeringExamples(),
+        triggeringExamples: AttributesRuleExamples.triggeringExamples()
+    )
+
+    public func validateFile(file: File) -> [StyleViolation] {
+        return validateTestableImport(file) +
+            validateFile(file, dictionary: file.structure.dictionary)
+    }
+
+    public func validateFile(file: File,
+                             kind: SwiftDeclarationKind,
+                             dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+
+        let attributeShouldBeOnSameLine: Bool?
+        if SwiftDeclarationKind.variableKinds().contains(kind) {
+            attributeShouldBeOnSameLine = true
+        } else if SwiftDeclarationKind.typeKinds().contains(kind) {
+            attributeShouldBeOnSameLine = false
+        } else if SwiftDeclarationKind.functionKinds().contains(kind) {
+            attributeShouldBeOnSameLine = false
+        } else {
+            attributeShouldBeOnSameLine = nil
+        }
+
+        let violations: [StyleViolation]
+        if let attributeShouldBeOnSameLine = attributeShouldBeOnSameLine {
+            violations = validateKind(file,
+                                      attributeShouldBeOnSameLine: attributeShouldBeOnSameLine,
+                                      dictionary: dictionary)
+        } else {
+            violations = []
+        }
+
+        return validateFile(file, dictionary: dictionary) + violations
+    }
+
+    private func validateTestableImport(file: File) -> [StyleViolation] {
+        let pattern = "@testable[\n]+\\s*import"
+        let excludingKinds = SyntaxKind.commentAndStringKinds()
```

Comment:
`@testable` is always a `source.lang.swift.syntaxtype.attribute.builtin` and `import` is always a `source.lang.swift.syntaxtype.keyword`, so you should use those rather than excluding comments and strings.